### PR TITLE
Video reviews

### PR DESCRIPTION
**Pull request by broadfootp**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    if @review.save
+      flash[:success] = "Your review has been added."
+      redirect_to video_path(@review.video)
+    else
+      flash.now[:danger] = "Your review contains errors."
+      @video = @review.video
+      @reviews = @video.reviews
+      render 'videos/show'
+    end
+  end
```
Comment (matthewbarram) [inline app/controllers/reviews_controller.rb]: Looks good.

File: app/models/video.rb [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+  has_many :reviews
   validates_presence_of :title, :description
   scope :search_by_title, -> (search_string) do
     return [] if search_string.blank?
     where("lower(title) LIKE ?", "%#{search_string.downcase}%").order("created_at")
   end
+
+  def average_rating
+    return nil if reviews.count == 0
+    avg = (reviews.map(&:rating).reduce(&:+)) / (reviews.count.to_f)
```
Comment (matthewbarram) [inline app/models/video.rb]: I suggest using the full work here.

File: app/views/videos/show.html.haml [hunk 3]
```diff
@@ -6,8 +6,40 @@
...
+%section.reviews.container
+  .row
+    .col-sm-10.col-sm-offset-1
+      = form_for [@video, Review.new] do |f|
+        %fieldset
+          .form-group
+            = f.label "Rate this video"
+            .row
+              .col-sm-3
+                = f.select "rating", [5,4,3,2,1].map { |n| [pluralize(n, "Star"), n]}, {}, class: "form-control"
```
Comment (matthewbarram) [inline app/views/videos/show.html.haml]: Use full work instead of `n` here

File: spec/controllers/videos_controller_spec.rb [hunk 4]
```diff
@@ -9,6 +9,14 @@
       expect(assigns(:video)).to eq(video)
     end
 
+    it "sets @reviews for authenticated users" do
+      session[:user_id] = Fabricate(:user).id
+      review1 = Fabricate(:review, video: video)
```
Comment (matthewbarram) [inline spec/controllers/videos_controller_spec.rb]: Consider making this a `let`

File: spec/controllers/videos_controller_spec.rb [hunk 5]
```diff
@@ -9,6 +9,14 @@
...
+      review2 = Fabricate(:review, video: video)
```
Comment (matthewbarram) [inline spec/controllers/videos_controller_spec.rb]: Same as above.

File: spec/models/video_spec.rb [hunk 6]
```diff
@@ -34,4 +35,33 @@
...
+      expect(video.average_rating).to be_nil
+    end
+    it "should return the review rating with one decimal if one review" do
+      review1 = Fabricate(:review, video: video, rating: 5)
+      expect(video.average_rating).to eq(5.0)
+    end
+    it "should return the average video rating to one decimal place" do
+      review1 = Fabricate(:review, video: video, rating: 1)
+      review2 = Fabricate(:review, video: video, rating: 5)
+      review3 = Fabricate(:review, video: video, rating: 4)
```
Comment (matthewbarram) [inline spec/models/video_spec.rb]: Move these to `let`.

File: app/views/videos/show.html.haml (hunk 3 above)
Comment (broadfootp) [inline app/views/videos/show.html.haml]: Thank you for your comments! I'm not sure I understand what do you mean by "use full work" here and in the comment above?

**Pull request merged**